src/rabbit_core_ff.erl

```diff
@@ -95,3 +103,19 @@ remove_explicit_default_bindings(FeatureName, Queues) ->
     [rabbit_binding:remove_default_exchange_binding_rows_of(Q)
      || Q <- Queues],
     ok.
+
+
+%% -------------------------------------------------------------------
+%% Virtual host metadata.
+%% -------------------------------------------------------------------
+
+virtual_host_metadata_migration(_FeatureName, _FeatureProps, enable) ->
+  Tab = rabbit_vhost,
+  rabbit_table:wait([Tab]),
+  Fun = fun vhost:upgrade/1,
+  case mnesia:transform_table(Tab, Fun, [virtual_host, limits, metadata]) of
+      {atomic, ok}      -> ok;
+      {aborted, Reason} -> {error, Reason}
+  end;
+virtual_host_metadata_migration(_FeatureName, _FeatureProps, is_enabled) ->
+  lists:member(metadata, mnesia:table_info(rabbit_vhost, attributes)).
```

Comment:
Just a style comment: the rest of RabbitMQ uses 4-space indentation.

Reply:
Likewise, instead of hard-coding the knowledge of `metadata`, please query the list of record fields and compare it to the table fields. Here is an example for the quorum queue migration:
```erlang
Fields = amqqueue:fields(amqqueue_v2),
mnesia:table_info(rabbit_queue, attributes) =:= Fields
```